src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc

```diff
@@ -329,10 +323,19 @@ std::optional<edm4eic::MutableCluster> CalorimeterClusterRecoCoG::reconstruct(co
   cl.addToShapeParameters( eigenValues_3D[0].real() ); // 3D x-y-z cluster width 1
   cl.addToShapeParameters( eigenValues_3D[1].real() ); // 3D x-y-z cluster width 2
   cl.addToShapeParameters( eigenValues_3D[2].real() ); // 3D x-y-z cluster width 3
-  //last 3 shape parameters are the components of the axis direction
-  cl.addToShapeParameters( axis_x );
-  cl.addToShapeParameters( axis_y );
-  cl.addToShapeParameters( axis_z );
+
+  if (axis_z != 0) {
```

Comment:
This is solely intended to catch singular cases where an eigenvector cannot be found, it seems (there is not div0 reason for it and only meaningfully test that the value is still at its initial value).

Are there more than the following two classes of these cases:
- two hits only,
- multiple hits where digitization and position quantization result in the perfect singularity of the cluster.

Presumably the second case is infrequent to the point of being negligible. The first may be numerous in the presence of low energy backgrounds.

I feel that adding the fallback gives clusters with two hits a different behavior which may lead to confusion. I also think that writing the exact zero in the intrinsic phi and theta may lead to confusion... I wish we could indicate validity of the eigenvalues somehow (without a special value, zero or otherwise, in the directions or their errors).

Reply:
The intention was to not handle trivial clusters, but handle detectors that don't provide 3d profile of the shower. I think, we should add a config flag to switch this instead.

Reply:
Ok, I see. There are then:
- trivial clusters because of number of hits,
- trivial clusters because of detector planarity,
- accidentally singular clusters.

I think the data model should define what to do for trivial clusters because of detector planarity. It seems that having this depend on a config flag (which is still hard to access in our output files) could cause confusion too.

Reply:
(Same comment as above about descoping applies here too.)

Reply:
Implemented. No handling of trivial clusters yet.